stdlib.sh

```diff
@@ -986,6 +986,38 @@ layout_pipenv() {
   export VIRTUAL_ENV
 }
 
+# Usage: layout poetry
+#
+# Similar to layout_python, but uses Poetry [1] to build a virtualenv from the
+# pyproject.toml [2] located in the same directory.
+#
+# Poetry by default will create a virtual environment under
+# {cache-dir}/virtualenvs where cache-dir equates to $XDG_CACHE_HOME/pypoetry
+# or use the {project-dir}/.venv directory when one is available.
+#
+# [1]: https://python-poetry.org/
+# [2]: https://peps.python.org/pep-0518/
+#
+layout_poetry() {
+    PYPROJECT_TOML="${PYPROJECT_TOML:-pyproject.toml}"
+    if [[ ! -f "$PYPROJECT_TOML" ]]; then
+        log_status "No pyproject.toml found. Executing \`poetry init\` to create a \`$PYPROJECT_TOML\` first."
```

Comment:
It seems `log_msg` does not exist but `log_status` does so have used that instead to avoid alarming the user.